app/components/session.service.js

```diff
@@ -0,0 +1,72 @@
+;(function () {
+  angular
+    .module("todo")
+    .service("SessionService", SessionService);
+
+  SessionService.$inject = ["$http", "API", "$state"];
+
+  function SessionService($http, API, $state) {
+    var self = this;
+    self.user = {
+      userView: "list"
+    };
+    self.getUser = getUser;
+    self.login = login;
+    self.registration = registration;
+    self.logout = logout;
+
+    self.getUser();
+
+    function getUser() {
+      $http.get(API.BASE + API.AUTH + API.INFO)
+        .then(function (response) {
+          self.user.data = response.data.user;
+          $state.go("groups.list");
+        })
+        .catch(function () {});
+    }
+
+    function login(user, successFn, errorFn) {
+      var result = $http.post(API.BASE + API.AUTH + API.LOGIN, user)
+        .then(function (response) {
+          callCallback(successFn);
+          self.getUser();
+          $state.go("groups.list");
+        })
+        .catch(function (response) {
+          callCallback(errorFn);
+        });
+    }
+
+    function registration(user, successFn, errorFn) {
+      var result = $http.post(API.BASE + API.REGISTRATION, user)
+        .then(function (response) {
+          if (response.data.success) {
+            callCallback(successFn);
+            self.getUser();
+            $state.go("groups.list");
+          } else {
+            callCallback(errorFn);
+          }
+        })
+        .catch(function () {})
+    }
+
+    function logout(successFn, errorFn) {
+      var result = $http.get(API.BASE + API.AUTH + API.LOGOUT)
+        .then(function () {
+          callCallback(successFn);
+          self.user.data = undefined;
+          $state.go("login");
+        })
+        .catch(function () {})
+    }
+  }
+
+  function callCallback(fn, args) {
```

Comment:
it's look like kind of hack, why do we need this? You already use promises, it's enough